scene/gui/button.cpp

```diff
@@ -672,6 +633,42 @@ void Button::_bind_methods() {
 	ADD_GROUP("BiDi", "");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "text_direction", PROPERTY_HINT_ENUM, "Auto,Left-to-Right,Right-to-Left,Inherited"), "set_text_direction", "get_text_direction");
 	ADD_PROPERTY(PropertyInfo(Variant::STRING, "language", PROPERTY_HINT_LOCALE_ID, ""), "set_language", "get_language");
+
+	BIND_THEME_ITEM(Theme::DATA_TYPE_STYLEBOX, Button, normal, "normal");
```

Comment:
The 3rd and 4th argument seem to always be the same (or there are cases where they aren't?). Maybe it's possible to merge them somehow?

Reply:
Unfortunately, they aren't always the same. In some cases members of `ThemeCache` are named a bit more explicitly, using a suffix like `*_style` or `*_icon`, for example. This can be changed so that they always match theme definition names, but I think it would make code of the classes slightly less easy to follow.

And we can't fix it by changing names of theme definitions, as that would be compatibility breaking.

Also this allows to do naughty but convenient things like storing x/y constants in a vector.

```cpp
BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, Label, font_shadow_offset.x, "shadow_offset_x");
BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, Label, font_shadow_offset.y, "shadow_offset_y");
```

There is another example in the editor code with the progress indicator where this would also be useful

```cpp
Ref<Texture2D> progress[8]{};

theme_cache.progress[0] = get_theme_icon("Progress1", SNAME("EditorIcons"));
theme_cache.progress[1] = get_theme_icon("Progress2", SNAME("EditorIcons"));
theme_cache.progress[2] = get_theme_icon("Progress3", SNAME("EditorIcons"));
...
```

Reply:
Maybe a separate macro for properties that would be repeated? 🤔

Reply:
I like the idea, that would make it easier to find such properties with mismatched name and cache identifier. Possibly to fix some of them in 5.0 when we can break compat.

Reply:
I don't mind adding a second set of macros for matching names (it should probably be the default then). My only concern is if it's going to look confusing because properties would be defined inconsistently depending on this name match. But, again, I don't mind giving it a go.